src/client_impl.cc

```diff
@@ -53,7 +35,9 @@ MixerClientImpl::MixerClientImpl(const MixerClientOptions &options)
   quota_cache_ =
       std::unique_ptr<QuotaCache>(new QuotaCache(options.quota_options));
 
-  deduplication_id_base_ = GenerateUUID();
+  if (options_.uuid_generate_func) {
```

Comment:
Adding that would essentially same as copying https://github.com/lyft/envoy/blob/master/source/common/runtime/runtime_impl.cc#L26 to here, because we need hex convert etc. I don't think it should be. We can make this option mandatory, so a constructor crash here.